racket/collects/net/http-client.rkt

```diff
@@ -237,6 +248,60 @@
   (http-conn-open! hc host-bs #:ssl? ssl? #:port port)
   hc)
 
+(define (http-conn-CONNECT-tunnel proxy-host proxy-port target-host target-port #:ssl? [ssl? #f])
+  (define hc (http-conn-open proxy-host #:port proxy-port #:ssl? #f))
+  (define connect-string (format "~a:~a" target-host target-port))
+  ; (log-net/url-info "http-conn-CONNECT-tunnel tunnel to ~s for ~s" connect-string (url->string url))
+  (http-conn-send! hc #:method "CONNECT" connect-string #:headers
+                   (list (format "Host: ~a" connect-string)
+                         "Proxy-Connection: Keep-Alive"
+                         "Connection: Keep-Alive"))
+
+  (let ((tunnel-status (http-conn-status! hc))
+        (tunnel-headers (http-conn-headers! hc)))
+    (unless (regexp-match "^HTTP[^ ]* +2" tunnel-status)
+      (error 'make-ports "HTTP CONNECT failed: ~a" tunnel-status)))
+
+  ;; SSL secure the ports
+  (match-define (http-conn _ _ _ t:to t:from _) hc)
+  (cond [(not ssl?) ; it's just a tunnel... no ssl
+         (define abandon-p (lambda (p) ((http-conn-abandon-p hc) p)))
+         (values ssl? t:from t:to abandon-p)]
+
+        [(osx-old-openssl?)
+         (http-conn-close! hc)
+         ;; osx-old-openssl? doesn't support ports->...ssl-ports
+         (error 'http-conn-open! "osx-old-openssl? does not support a ports->...ssl-ports function")]
```

Comment:
Oh I see... I thought you were taking about the "HTTP CONNECT failed" error,
which is bang in the middle of the extract above. No... you're taking about the
`osx-old-openssl?`, innit?

I'll remove the `cond` clause; I think I misread what `osx-old-openssl?` does
for us. And we'll let them run against an error in the openssl libraries.
(Which, from my experience, will be utterly incomprehensible)